examples/multisig_plain/lib.rs

```diff
@@ -0,0 +1,837 @@
+            let mut contract = submit_transaction();
+            contract.cancel_transaction(0);
+        }
+
+        #[test]
+        fn confirm_transaction_works() {
+            let mut contract = submit_transaction();
+            let accounts = default_accounts();
+            set_sender(accounts.bob);
+            contract.confirm_transaction(0);
+            assert_eq!(test::recorded_events().count(), 3);
+            contract.confirmations.get(&(0, accounts.bob)).unwrap();
+            assert_eq!(contract.confirmations.len(), 2);
+            assert_eq!(*contract.confirmation_count.get(&0).unwrap(), 2);
+        }
+
+        #[test]
+        fn confirm_transaction_already_confirmed() {
+            let mut contract = submit_transaction();
+            let accounts = default_accounts();
+            set_sender(accounts.alice);
+            contract.confirm_transaction(0);
+            assert_eq!(test::recorded_events().count(), 2);
+            contract.confirmations.get(&(0, accounts.alice)).unwrap();
+            assert_eq!(contract.confirmations.len(), 1);
+            assert_eq!(*contract.confirmation_count.get(&0).unwrap(), 1);
+        }
+
+        #[test]
+        #[should_panic]
+        fn confirm_transaction_noowner_fail() {
+            let mut contract = submit_transaction();
+            set_from_noowner();
+            contract.confirm_transaction(0);
+        }
+
+        #[test]
+        fn revoke_transaction_works() {
+            let mut contract = submit_transaction();
+            let accounts = default_accounts();
+            set_sender(accounts.alice);
+            contract.revoke_confirmation(0);
+            assert_eq!(test::recorded_events().count(), 3);
+            assert!(contract.confirmations.get(&(0, accounts.alice)).is_none());
+            assert_eq!(contract.confirmations.len(), 0);
+            assert_eq!(*contract.confirmation_count.get(&0).unwrap(), 0);
+        }
+
+        #[test]
+        fn revoke_transaction_no_confirmer() {
+            let mut contract = submit_transaction();
+            let accounts = default_accounts();
+            set_sender(accounts.bob);
+            contract.revoke_confirmation(0);
+            assert_eq!(test::recorded_events().count(), 2);
+            assert!(contract.confirmations.get(&(0, accounts.alice)).is_some());
+            assert_eq!(contract.confirmations.len(), 1);
+            assert_eq!(*contract.confirmation_count.get(&0).unwrap(), 1);
+        }
+
+        #[test]
+        #[should_panic]
+        fn revoke_transaction_noowner_fail() {
+            let mut contract = submit_transaction();
+            let accounts = default_accounts();
+            set_sender(accounts.django);
+            contract.revoke_confirmation(0);
+        }
+
+        #[test]
+        fn execute_transaction_works() {
+            // Execution of calls is currently unsupported in off-chain test.
+            // Calling execute_transaction panics in any case.
```

Comment:
So sad. :(

Reply:
Maybe we can change it from panic to "nothing at all". Then I can at least test if it tried to execute or failed for some other reason.

Reply:
We could try but on the other hand I don't think that this feature is even on the horizon.

Reply:
I guess we leave it that way then and re-iterate on the example later.